data/src/main/java/com/linkedin/data/schema/compatibility/AnnotationCompatibilityChecker.java

```diff
@@ -0,0 +1,133 @@
+ *  In this checker, it will call the annotationCompatibilityCheck method which provide in the SchemaAnnotationHandler
+ *  by user to do the annotation compatibility check.
+ *
+ * @author Yingjie Bi
+ */
+public class AnnotationCompatibilityChecker
+{
+  private static final Logger _log = LoggerFactory.getLogger(AnnotationCompatibilityChecker.class);
+
+  /**
+   * Check the annotation compatibilities
+   * process olderSchema and newerSchema in the SchemaAnnotationProcessor to get the resolved result with resolvedProperties.
+   * then using the resolvedProperties to do the annotation compatibility check.
+   * @param olderSchema
+   * @param newerSchema
+   * @param handlers SchemaAnnotationHandler list
+   * @return List<AnnotationCompatibilityResult>
+   */
+  public static List<AnnotationCompatibilityResult> checkPegasusSchemaCompatibility(DataSchema olderSchema, DataSchema newerSchema,
+      List<SchemaAnnotationHandler> handlers)
+  {
+    SchemaAnnotationProcessor.SchemaAnnotationProcessResult olderSchemaResult = processSchemaAnnotation(olderSchema, handlers);
+    SchemaAnnotationProcessor.SchemaAnnotationProcessResult newerSchemaResult = processSchemaAnnotation(newerSchema, handlers);
+    Map<PathSpec, Pair<SchemaAnnotationHandler.AnnotationCheckContext, Map<String, Object>>> olderResolvedPropertiesMap
+        = getNodeToResolvedProperties(olderSchemaResult);
+    Map<PathSpec, Pair<SchemaAnnotationHandler.AnnotationCheckContext, Map<String, Object>>> newerResolvedPropertiesMap
+        = getNodeToResolvedProperties(newerSchemaResult);
+
+    return getCompatibilityResult(olderResolvedPropertiesMap, newerResolvedPropertiesMap, handlers);
+  }
+
+  /**
+   * Iterate the nodeToResolverPropertiesMap, if olderResolvedPropertiesMap and newerResolvedPropertiesMap contain same pathSpec,
+   * calling annotationCompatibilityCheck api which is provided in the SchemaAnnotationHandler to do the annotation compatibility check.
+   * @param olderResolvedPropertiesMap
+   * @param newerResolvedPropertiesMap
+   * @param handlers SchemaAnnotationHandler list
+   * @return List<AnnotationCompatibilityResult>
+   */
+  static List<AnnotationCompatibilityResult> getCompatibilityResult(Map<PathSpec, Pair<SchemaAnnotationHandler.AnnotationCheckContext,
+      Map<String, Object>>> olderResolvedPropertiesMap, Map<PathSpec, Pair<SchemaAnnotationHandler.AnnotationCheckContext,
+      Map<String, Object>>> newerResolvedPropertiesMap, List<SchemaAnnotationHandler> handlers)
+  {
+    List<AnnotationCompatibilityResult> results = new ArrayList<>();
+    for (SchemaAnnotationHandler handler: handlers)
+    {
+      olderResolvedPropertiesMap.entrySet().stream().forEach(e ->
+          {
+            String annotationNamespace = handler.getAnnotationNamespace();
+            PathSpec pathSpec = e.getKey();
+            if (newerResolvedPropertiesMap.containsKey(pathSpec))
```

Comment:
How does the handler handle these Else branch cases?  Maybe add some brief comments

Reply:
Updated the comments and also added logic to handle else branch cases.